From 99d2462b80435b91a35236639b909eebee432126 Mon Sep 17 00:00:00 2001
From: Tony Murray <murraytony@gmail.com>
Date: Fri, 29 Oct 2021 20:34:24 -0500
Subject: [PATCH] Fix widget title injection vulnerability (#13452)

Prevent html/js code from being injected into widget titles.
---
 app/Http/Controllers/OverviewController.php       | 2 +-
 app/Http/Controllers/Widgets/WidgetController.php | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/app/Http/Controllers/OverviewController.php b/app/Http/Controllers/OverviewController.php
index 000518804cd..bdc4d911e1c 100644
--- a/app/Http/Controllers/OverviewController.php
+++ b/app/Http/Controllers/OverviewController.php
@@ -103,7 +103,7 @@ public function default(Request $request)
         $data = serialize(json_encode($data));
         $dash_config = unserialize($data);
         $hide_dashboard_editor = UserPref::getPref($user, 'hide_dashboard_editor');
-        $widgets = Widget::select('widget_id', 'widget_title')->orderBy('widget_title')->get();
+        $widgets = Widget::select(['widget_id', 'widget_title'])->orderBy('widget_title')->get();
 
         $user_list = [];
         if ($user->can('manage', User::class)) {
diff --git a/app/Http/Controllers/Widgets/WidgetController.php b/app/Http/Controllers/Widgets/WidgetController.php
index a58f8e073ba..34d6b8ed7df 100644
--- a/app/Http/Controllers/Widgets/WidgetController.php
+++ b/app/Http/Controllers/Widgets/WidgetController.php
@@ -142,7 +142,7 @@ private function formatResponse($view, $title, $settings, $status = 'ok')
 
         return response()->json([
             'status' => $status,
-            'title' => __($title),
+            'title' => htmlentities(__($title)),
             'html' => $html,
             'show_settings' => $show_settings,
             'settings' => $settings,
